src/x3d/x3dload.pas

```diff
@@ -351,13 +334,78 @@ function LoadNode(const URL: string;
     Result := LoadGLTF(URL)
   else
 
+  if (MimeType = 'image/starling-texture-atlas') or
+     (URIMimeType(URIDeleteAnchor(URL, true)) = 'image/starling-texture-atlas') then
+    Result := LoadStarlingTextureAtlas(URL)
+  else
+
+  if (MimeType = 'application/x-plist') or
+     (URIMimeType(URIDeleteAnchor(URL, true)) = 'application/x-plist') then
+    Result := LoadCocos2d(URL)
+  else
+
+  { Support for simple graphics images like PNG }
+  if URIMimeExtensions.ContainsKey(ExtractFileExt(URIDeleteAnchor(URL, true))) then
```

Comment:
This checks that URL extension is defined somewhere in our MIME database in `URIMimeExtensions`. 

While only images are registered there now (so in practice this does the job OK) but in general various units (in CGE or user code) could register other file types in `URIMimeExtensions`, not necessarily images.

Rejected solution 1: The right array to browse is `ImageFormatInfos` inside `CastleImage`. Rejected: but it is internal in that unit, and I would prefer to keep it internal (to allow more flexible implementation with registered image formats in the future).

Rejected solution 2: Expose in `CastleImages` a public `ImageMimeExtensions` (function in interface, exposing an internal singleton, created inside the unit). `ImageMimeExtensions` would be populated only in `RegisterMimeTypes` in `src/images/castleimages_file_formats.inc`, with the same content as `URIMimeExtensions`. Rejected: This could be modified by user code.

Good solution: I would say to expose in `CastleImages` dedicated functions for this. They seem to be generally useful, so it makes sense to expose them:

- Expose in `CastleImages` a function `IsImageMimeType(const MimeType: String): Boolean` (that internally accesses `ImageFormatInfos`, iterating over every format, and in each format iterating over every `TImageFormatInfo.MimeTypes` item).

- For the ReadImageExtensions below, similarly: make in `CastleImages` function `AllImagesExtensions` that would return the desired String. I have a sketch of such function:

```pascal
function AllImagesExtensions: String;
var
  Format: TImageFormat;
  ExtIndex: Integer;
begin
  Result := '';
  for Format := Low(Format) to High(Format) do
    for ExtIndex := 1 to ImageFormatInfos[Format].ExtsCount do
      Result := SApppendPart(Result, ';', '*.' + ImageFormatInfos[Format].Exts[ExtIndex];
end;
```

Reply:
For the `ReadImageExtensions` I think I found better solution. I propose using `LoadImage_FileFilters` and add `function FilterString :String` to `TFileFilterList`. This aproach also makes `TFileFilterList` more complete. What do you think about that?

Reply:
Of course, we needed a TFilterList.AllExtensions: String function that omitted * and *. *.